arch/posix/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+zephyr_cc_option_ifdef(CONFIG_LTO -flto)
+
+zephyr_ld_option_ifdef(CONFIG_COVERAGE
+	-lgcov
+	)
+zephyr_compile_options(
+  -m32
+  -MMD -MP -D_POSIX_C_SOURCE=199309
```

Comment:
zephyr_compile_definitions(_POSIX_C_SOURCE=199309)

Reply:
(Or perhaps even better just put this in the cheat file)

Reply:
I don't think it would be good to put it in the cheat file (this macro is meant for the native standard headers) => I like more your first proposal